src/Take.Elephant.Kafka/PartitionedKafkaReceiver.cs

```diff
@@ -0,0 +1,177 @@
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using Confluent.Kafka;
+
+namespace Take.Elephant.Kafka
+{
+    public class PartitionedKafkaReceiver<TKey, TEvent> : IEventStreamConsumer<TKey, TEvent>, IOpenable, ICloseable, IDisposable
```

Comment:
This class has literally the same implementation as `KafkaReceiverQueue`. 

I know you just purposefully separated the two (queue and stream) here from the old PR since they are different concepts, as per the suggestion of @andrebires, however I think we could find a way to share this code.

Even conceptually, I think it is reasonable to share them, as our Kafka Queue implementation is obviously a queue implemented on top of a stream; It's basically a stream with queue semantics (only because it's not FIFO (as we won't guarantee ordering), we cannot say it's _literally_ a stream with queue semantics, so, even in that sense, it leans more on the stream side than on the queue side). 

So, perhaps we can make `KafkaReceiverQueue` inherit from this? Or maybe both inherit from a new class?